test/snippet/alphabet/nucleotide/dna4.cpp

```diff
@@ -7,23 +7,21 @@ int main()
 
     seqan3::dna4 my_letter{'C'_dna4};
 
-    my_letter.assign_char('F'); // Characters other than IUPAC characters are implicitly converted to `A`.
-    if (my_letter.to_char() == 'A')
-        seqan3::debug_stream <<  "yeah\n"; // "yeah";
+    my_letter.assign_char('F'); // characters other than IUPAC characters are implicitly converted to A.
+    seqan3::debug_stream << my_letter; // prints "F"
 
-    // IUPAC characters are implicitly converted to their best fitting representative.
-    my_letter.assign_char('R'); // my_letter == 'A'_dna4;
-    my_letter.assign_char('Y'); // my_letter == 'C'_dna4;
-    my_letter.assign_char('S'); // my_letter == 'C'_dna4;
-    my_letter.assign_char('W'); // my_letter == 'A'_dna4;
-    my_letter.assign_char('K'); // my_letter == 'G'_dna4;
-    my_letter.assign_char('M'); // my_letter == 'A'_dna4;
-    my_letter.assign_char('B'); // my_letter == 'C'_dna4;
-    my_letter.assign_char('D'); // my_letter == 'A'_dna4;
-    my_letter.assign_char('H'); // my_letter == 'A'_dna4;
-    my_letter.assign_char('V'); // my_letter == 'A'_dna4;
+    // IUPAC characters are implicitly converted to their best fitting representative
+    seqan3::debug_stream << my_letter.assign_char('R'); // prints "A"
+    seqan3::debug_stream << my_letter.assign_char('Y'); // prints "C"
+    seqan3::debug_stream << my_letter.assign_char('S'); // prints "C"
+    seqan3::debug_stream << my_letter.assign_char('W'); // prints "A"
+    seqan3::debug_stream << my_letter.assign_char('K'); // prints "G"
+    seqan3::debug_stream << my_letter.assign_char('M'); // prints "A"
+    seqan3::debug_stream << my_letter.assign_char('B'); // prints "C"
+    seqan3::debug_stream << my_letter.assign_char('D'); // prints "A"
+    seqan3::debug_stream << my_letter.assign_char('H'); // prints "A"
+    seqan3::debug_stream << my_letter.assign_char('V'); // prints "A"
```

Comment:
```suggestion
    seqan3::debug_stream << 'R'_my_dna4; // prints "A"
    seqan3::debug_stream << 'Y'_my_dna4; // prints "A"
    seqan3::debug_stream << 'S'_my_dna4; // prints "A"
    seqan3::debug_stream << 'W'_my_dna4; // prints "A"
    seqan3::debug_stream << 'K'_my_dna4; // prints "A"
    seqan3::debug_stream << 'M'_my_dna4; // prints "A"
    seqan3::debug_stream << 'B'_my_dna4; // prints "A"
    seqan3::debug_stream << 'D'_my_dna4; // prints "A"
    seqan3::debug_stream << 'H'_my_dna4; // prints "A"
    seqan3::debug_stream << 'V'_my_dna4; // prints "A"
```
Not sure, if this is documentation is helpful since the example with F would already indicate what happens.

Reply:
No this is the dna4.cpp ! not the custom my_dna4. 
I added this in dna4.cpp because I think it is helpful to see what "our" dna4 does with IUPAC characters

Reply:
Ah ok, I mixed that because of my_letter some how.